src/Mod/Spreadsheet/Gui/Command.cpp

```diff
@@ -197,9 +197,36 @@ void CmdSpreadsheetImport::activated(int iMsg)
     if (!fileName.isEmpty()) {
         std::string FeatName = getUniqueObjectName("Spreadsheet");
         Sheet * sheet = freecad_dynamic_cast<Sheet>(App::GetApplication().getActiveDocument()->addObject("Spreadsheet::Sheet", FeatName.c_str()));
+        ParameterGrp::handle group = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Spreadsheet");
+        QString delimiter = QString::fromStdString(group->GetASCII("ImportExportDelimiter","tab"));
+        QString quoteChar = QString::fromStdString(group->GetASCII("ImportExportQuoteCharacter","\""));
+        QString escapeChar = QString::fromStdString(group->GetASCII("ImportExportEscapeCharacter","\\"));
+        group->SetASCII("ImportExportDelimiter", Base::Tools::toStdString(delimiter).c_str()); //create parameters that do not already exist for user convenience
+        group->SetASCII("ImportExportQuoteCharacter", Base::Tools::toStdString(quoteChar).c_str());
+        group->SetASCII("ImportExportEscapeCharacter", Base::Tools::toStdString(escapeChar).c_str());

```

Comment:
I don't like much this kind of code that is useful only once but executed every time.
To me, either it's only a parameter then document it in the Fine-tuning wiki page, or it's a preference then the parameter creation will be handled there.
But it may be only a personal opinion. :)

Reply:
While in principle I agree, in actual practice it is VERY nice to have these variables automatically appear in the user.cfg and the Edit Parameters screen. Something like _half_ of #4787 would be unnecessary if we did this for all preferences!

Reply:
@chennes In this case I'd be more in favor to have a mechanism at startup (or workbench load) that takes care of that only once. ;)

Reply:
Right, I'd be fine with this moving/copying to the constructor (or loader, etc.) and only doing an access here, rather than the set.

Reply:
I'm not sure of that as it would mean to duplicate the location where default value is defined (or using some kind of a #DEFINE to store it). IMO it would be too risky to get the values without setting the default in case user inadvertently delete them...

Reply:
I'm not sure what you're suggesting, then. We basically _have_ to do the get here, in case the preference changes after the WB is loaded. And there's no risk to doing a "get" when the user has deleted the preference, the system will just return the default value that's given as part of the "get" command:
```
QString delimiter = QString::fromStdString(group->GetASCII("ImportExportDelimiter","tab"));
QString quoteChar = QString::fromStdString(group->GetASCII("ImportExportQuoteCharacter","\""));
QString escapeChar = QString::fromStdString(group->GetASCII("ImportExportEscapeCharacter","\\"));
```
The big downside is that those three lines have to be duplicated, and if one of them ever changes, you have to make sure to repeat that change wherever you are doing the set of the default. So the efficiency gain of not setting them every time comes at a price of code duplication. I don't feel strongly about it, neither solution is actually "good," that's just how our preference system works.